codecov.yml

```diff
@@ -23,15 +23,64 @@ coverage:
       default:
         branches:
           - develop
+        target: 90%
         informational: true
         only_pulls: true
         paths:
-          - "nncf/onnx"
-          - "nncf/common"  # extend this once we collect coverage reports for more than just onnx and common part of precommit
+          - "nncf"
 
 comment:
-  layout: "diff, flags, files"
+  layout: "reach, diff, files, flags, components"
   require_changes: false
 
   require_head: false
   require_base: false
+
+flag_management:
+  # Flag coverage percentage seems to show the "percentage of lines under the flag path covered as reported ONLY
+  # by the upload with the corresponding flag", so e.g. for COMMON the flag coverage percentage will report the
+  # percentage of common code tested ONLY by the common tests, and e.g. not by backend-specific precommit parts
+  # (which also run common code and are therefore indirectly providing coverage). Ideally each flag-specific path
+  # would be described below with the corresponding flag and provide valuable information on whether the test code base
+  # is written efficiently, e.g. that the backend-specific tests predominantly validate backend-specific code and the
+  # common tests completely cover the common code on their own. However, if we set all flags with paths here, then the
+  # total repo coverage percentage will sink, because codecov currently reports the overall coverage based on the union
+  # of the "flag" coverages - not the "component" coverages (see below) - and currently NNCF's precommit tests are
+  # biased toward validating common code via backend-specific tests. In the future the tests will be gradually
+  # refactored to have more "locality" in what each precommit section tests.
+  individual_flags:
+    - name: COMMON
+      paths:
+        - nncf/common
+        - nncf/quantization
+
+component_management:
+  # In contrast to the "flag" coverage above, the "component" display seems to calculate percentage based on the
+  # coverage information from ALL uploads for the code in the specified path. With this, the "component" coverage
+  # percentage is a better representation of what sub-paths in the NNCF code base are covered with at least one test,
+  # without distinction whether the test was run in the
+  individual_components:
+    - component_id: common
+      name: common
+      paths:
+        - nncf/common
+    - component_id: torch
+      name: torch
+      paths:
+        - nncf/torch
+    - component_id: tensorflow
+      name: tensorflow
+      paths:
+        - nncf/tensorflow
+    - component_id: onnx
+      name: onnx
+      paths:
+        - nncf/onnx
+    - component_id: openvino
+      name: openvino
+      paths:
+        - nncf/openvino
+    - component_id: quantization
+      name: common_ptq
```

Comment:
The current division by components is not correct.  Now `common_ptq`  collects files for all backends.
Please split all files to components by backends (conditions was discussed in https://github.com/openvinotoolkit/nncf/pull/1799)

Reply:
If we are talking about the Codecov term for a `"component"`, then it is absolutely arbitrary what subset of files we choose to be a component. I propose the folder-based components in the form they are in now. The component "common_ptq" is now reflecting the coverage for the PTQ component of the NNCF, for all backends. I will rename it to "ptq" since this is not really common, as you pointed out.

I have also just now extended the backend-specific "components" with corresponding files based on a glob/regex. There is no easy way to test whether they matched correctly, though. It doesn't seem to be documented, or possible, to specify `ignore` sections for "component" file paths, which means that the `common` "component" coverage will not reflect PTQ common code coverage, and that the `ptq` "component" coverage will now be calculated based off a set of files that overlaps with each backend-specific "component". 

By the way, all of these troubles, including the effort spent on #1799, follow from the original bad decision to break existing file structure consistency and store backend-specific PTQ code in `*_backend.py` instead of at `nncf/*backend*` like it was done before the POT started to be merged in, and I pointed out this decision as bad from the start (@alexsu52 @KodiaqQ FYI). We will have such issues with every future tool that works on our code base and where we need component-level breakdown of the results.

Reply:
>  It doesn't seem to be documented, or possible, to specify ignore sections for "component" file paths

https://docs.codecov.com/docs/path-regexp

Reply:
Looks like I missed it, thanks. Pushed the version with ignores in the common component.